src/main/java/com/ase/restservice/service/StockService.java

```diff
@@ -21,9 +23,24 @@ public class StockService implements StockServiceI {
    *
    * @param stock Stock
    * @return Created stock
+   * @throws ResourceNotFoundException if the stock ID is invalid
    */
-  public Stock createStock(Stock stock) {
+  public Stock createStock(Stock stock) throws ResourceNotFoundException {
     // TODO: Throw exception if stock already exists
+
+    // check if stock ID is valid
+    String stockId = stock.getStockId();
+    try {
+        yahoofinance.Stock apiStock = YahooFinance.get(stockId);
```

Comment:
After we create the FinanceService, we can use that wrapper rather than directly calling YahooFinance here and everywhere. However, I think this stockService.createStock() should still just create the stock with the Stock object that is passed in. Here, we are overriding the given stockPrice and retrieving a live one. We can move this functionality to financeService.createStockFromSymbol(). This method would take in a string of stock symbol (e.g. "AMZN"), then call financeService.getStockPriceBySymbol(), and then use that price along with the stock symbol to create a Stock object, and eventually call stockService.createStock() with it.